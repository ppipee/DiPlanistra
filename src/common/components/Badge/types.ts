export enum BadgeVariant {
	Outlined = 'outlined',
	Filled = 'filled',
}

export type BadgeVariantTypes = BadgeVariant | 'filled' | 'outlined'

export interface BadgeProps {
	$variant?: BadgeVariantTypes
	$color: string
	$secondColor?: string
}
