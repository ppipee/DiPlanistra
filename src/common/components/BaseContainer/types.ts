import { ScreenName } from 'common/styles/utils/viewport/screenSizes'

export interface BaseContainerProps {
	$screen?: ScreenName
	$padding?: string
	$paddingMobile?: string
	$margin?: string
	$marginMobile?: string
}
