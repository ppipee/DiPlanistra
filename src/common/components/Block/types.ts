export interface BlockProps {
	$padding?: string
	$borderRadius?: string
	$margin?: string
}
