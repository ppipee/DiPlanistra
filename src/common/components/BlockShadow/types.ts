export enum Shadow {
	Default = 'default',
	Hard = 'hard',
}

type ShadowType = Shadow | Shadow.Default | Shadow.Hard

export interface BlockShadowProps {
	$variant?: ShadowType
}
