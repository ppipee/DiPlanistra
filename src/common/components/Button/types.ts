export interface ButtonProps {
	$size?: SizeType
	$variant?: VariantType
	$responsive?: boolean
	$disable?: boolean
	$color: string
	$secondaryColor?: string
	$shadow?: boolean
	$border?: BorderType
}

type SizeType = ButtonSizes | 'small' | 'default'

export enum ButtonSizes {
	Small = 'small',
	Default = 'default',
}

type VariantType = ButtonVariants | 'outlined' | 'filled'

export enum ButtonVariants {
	Outlined = 'outlined',
	Filled = 'filled',
}

type BorderType = ButtonBorders | 'default' | 'curve'

export enum ButtonBorders {
	Default = 'default',
	Curve = 'curve',
}
