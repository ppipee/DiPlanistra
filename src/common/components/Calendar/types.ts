export interface CalenderDateProps {
	startDate?: Date
	endDate?: Date
}

export interface CalendarDisplayStateProps {
	year: number
	month: number
}

export interface CalendarDisplayRange {
	startRange?: CalendarDisplayStateProps
	endRange?: CalendarDisplayStateProps
}

export interface CalendarCellProps {
	$inBetween?: boolean
	$selected?: boolean
	$variant?: CalendarCellVariant
	$isFirstCell?: boolean
	$isLastCell?: boolean
	$isDateSelectable?: boolean
}

export type CalendarCellVariant = 'currentMonth' | 'otherMonth' | 'currentDate'
