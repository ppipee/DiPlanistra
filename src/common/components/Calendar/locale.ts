export const CALENDAR_START_DATE_TITLE = ['เริ่มต้น', 'Start Date']
export const CALENDAR_END_DATE_TITLE = ['สิ้นสุด', 'End Date']
