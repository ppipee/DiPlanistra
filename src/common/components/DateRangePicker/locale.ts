export const PICKER_YESTERDAY = ['เมื่อวาน', 'Yesterday']
export const PICKER_LAST_7_DAYS = ['7 วันที่แล้ว', 'Last 7 Days']
export const PICKER_LAST_14_DAYS = ['14 วันที่แล้ว', 'Last 14 Days']
export const PICKER_LAST_30_DAYS = ['30 วันที่แล้ว', 'Last 30 Days']
export const PICKER_THIS_MONTH = ['เดือนนี้', 'This Month']
export const CUSTOM_PICKER = ['กำหนดเอง', 'Custom']
