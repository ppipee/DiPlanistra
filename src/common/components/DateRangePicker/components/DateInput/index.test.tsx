import React from 'react'

import { fireEvent, render } from '@testing-library/react'

import { createI18nWrapper } from 'core/locale/testUtils'
import { Locale } from 'core/locale/types'

import * as DateInputType from '.'

describe('Test DateInput', () => {
	const useDateRangeContextSpy = jest.fn()
	const openSpy = jest.fn()
	const baseDateInputSpy = jest.fn()
	const getDisplayDateSpy = jest.fn()

	jest.doMock('common/components/DateRangePicker/utils/getDisplayDate/index', () => getDisplayDateSpy)
	jest.doMock('common/components/DateRangePicker/context', () => ({
		useDateRangeContext: useDateRangeContextSpy,
	}))
	jest.doMock('common/components/DateRangePicker/components/BaseDateInput', () => (props: any) => {
		baseDateInputSpy(props)
		return <div {...props}>{props.children}</div>
	})

	const { default: DateInput } = require('.') as typeof DateInputType

	beforeEach(() => {
		getDisplayDateSpy.mockReturnValue('Mock Display Date')
	})

	afterEach(() => {
		jest.resetAllMocks()
	})

	it('should render DateInput correctly', () => {
		useDateRangeContextSpy.mockReturnValue({
			date: 'MOCK DATE',
			controllerIndex: 1,
		})

		const { container } = render(<DateInput open={openSpy} />, {
			wrapper: createI18nWrapper(Locale.th),
		})

		expect(container.firstChild).toMatchSnapshot()
	})

	it('should perform DateInput correctly', () => {
		useDateRangeContextSpy.mockReturnValue({
			date: 'MOCK DATE',
			controllerIndex: 1,
		})

		const { container } = render(<DateInput open={openSpy} />, {
			wrapper: createI18nWrapper(Locale.th),
		})

		fireEvent.click(container.firstChild)

		expect(openSpy).toBeCalledTimes(1)
		expect(useDateRangeContextSpy).toBeCalledTimes(1)
		expect(baseDateInputSpy.mock.calls[0][0].value).toBe('Mock Display Date')

		expect(getDisplayDateSpy).toBeCalledTimes(1)
		expect(getDisplayDateSpy).toBeCalledWith('MOCK DATE')
	})
})
