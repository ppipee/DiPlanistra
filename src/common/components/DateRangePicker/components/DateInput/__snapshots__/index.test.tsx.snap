// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test DateInput should render DateInput correctly 1`] = `
<div
  class="styled__DateInputWrapper-vaoKq capbpl"
>
  <div
    value="Mock Display Date"
  />
</div>
`;
